### ISSUE TITLE
Multiple subscriptions per 1 client - same domain "path" - Unable to set certificate name

### ISSUE THREAD

**Issue by mnemotiv** [plesk/letsencrypt-plesk#51]
OS: Debian 8.2‬
Plesk version: 12.5.30 Update #18, last updated at Dec 31, 2015 06:26 AM
Let's Encrypt: 1.2, release 1

How to reproduce:

- create customer and subscription with existing root-domain, eg. example.com
- create additional subscription for the same customer, but with sub-domain for the same root-domain, eg. s0.example.com (I need this because I've set different shared IP address for this sub-domain - cannot set different shared IP addresses per domain/subdomain on a single subscription)
- try to create/renew certificate for root-domain or sub-domain (doesn't care which one) - result: you get errors below - seems like bug in ioncube-encoded Plesk extension in this reproducable situation...
- if I delete one of the subscriptions - everything is ok, seems like extension can't get list of domains correctly or something like that.

```
[31-Dec-2015 11:56:04 Europe/Helsinki] PHP Warning: Invalid argument supplied for foreach(); File: /opt/psa/admin/plib/modules/letsencrypt/library/Helper/DomainManipulator.php, Line: 26
[2015-12-31 11:56:04] ERR [1] PHP Warning: Invalid argument supplied for foreach(); File: /opt/psa/admin/plib/modules/letsencrypt/library/Helper/DomainManipulator.php, Line: 26
[2015-12-31 11:56:04] ERR [panel] Let&#039;s Encrypt SSL certificate installation failed: Failed letsencrypt execution: 2015-12-31 11:55:58,032:WARNING:letsencrypt.cli:Root (sudo) is required to run most of letsencrypt functionality.<br />
Install certificate failure: Unable to set certificate name :<br />
IMPORTANT NOTES:<br />
 - Congratulations! Your certificate and chain have been saved at /opt<br />
   /psa/var/modules/letsencrypt/etc/live/example.com/fullchain.pem.<br />
   Your cert will expire on 2016-03-30. To obtain a new version of the<br />
   certificate in the future, simply run Let&#039;s Encrypt again.
```

**Comment by xgin** [comment]
The issue seems similar to #32: the installation error is "Unable to set certificate name :"
The warnings are actually the problem in the extension, but is not a blocker.

The root cause of your failure seems database inconsistency in Plesk: the previous certificate with the same name had not been removed.
I guess you created a subdomain, installed a certificate, and then removed it, did not you?
Check the certificate is present in Plesk database:
```plesk db "select id, name from certificates where name = 'Lets Encrypt s0.example.com'"```

Check if there is any inconsistency in the database:
```
plesk db "select c.id, c.name, r.rep_id, d.name from certificates c left join Repository r on (c.id = r.component_id) left join domains d on (r.rep_id = d.cert_rep_id) where r.rep_id not in (select val from misc where param = 'cert_rep_id') or r.rep_id is null"
+----+-----------------------------------------+--------+-----------------------+
| id | name                                    | rep_id | name                  |
+----+-----------------------------------------+--------+-----------------------+
|  3 | ekazakov-2014                           |      2 | ekazakov.ru           |
|  5 | ekazakov-2015                           |      2 | ekazakov.ru           |
| 27 | Lets Encrypt ekazakov.ru                |      2 | ekazakov.ru           |
| 26 | Lets Encrypt le.testme.ekazakov.ru      |      3 | le.testme.ekazakov.ru |
| 15 | Lets Encrypt new-sub.testme.ekazakov.ru |      4 | NULL                  |
| 28 | Lets Encrypt drupal.ekazakov.ru         |      5 | drupal.ekazakov.ru    |
+----+-----------------------------------------+--------+-----------------------+
```
Here is a record with **NULL** - it is orphaned. We should clean it, because Plesk is not able to install a new certificate with the same name and also not able to remove it without link to any domain.
```
plesk db "delete r.* from Repository r inner join certificates c on (r.component_id = c.id) where c.name = 'Lets Encrypt new-sub.testme.ekazakov.ru'"
plesk db "delete from certificates where name = 'Lets Encrypt new-sub.testme.ekazakov.ru'"
```

Sum up: there is an issue in Plesk caused inconsistency after subdomain removal. I hope it will be fixed in one of the nearest updates. The inconsistency blocks an installation of new certificates for the same domain name. Fix the inconsistency manually.

**Comment by mnemotiv** [comment]
@xgin that's true - I've removed/add that subdomain multiple times without removing certificate manualy.

thanks for help! I hope the fix will come soon.

**Comment by xgin** [comment]
@mnemotiv take into account, an update is hardly going to remove data from your database, even an inconsistent! The purpose of the fix is preventing such issues.
You should repair your database by yourself, according to the queries above.
Let me know if any help is required.

**Comment by xgin** [comment]
The referenced issue was fixed in Plesk 12.5 update 25:
http://docs.plesk.com/release-notes/12.5/change-log/#12530-mu25
> If a domain with a Let’s Encrypt certificate was removed, another Let’s Encrypt certificate could not be installed on the domain if it was re-created. (PPPM-4166)
